DESIGN.md

```diff
@@ -2,95 +2,123 @@
+downstream (default name: `zipkin-trace`)
 
-When search requests are received, span index is used to search for trace ids. After a list is 
-retrieved, trace DAG is retrieved from trace state store.
+**Dependencies**
 
-##### Get Dependencies
+Once `traces` are emitted downstream as part of the initial processing, dependency links are evaluated
+on each trace, and emitted on another topic (default name: `zipkin-dependency`) for further metric aggregation.
 
-After `spans` are aggregated into traces, traces are processed to collect dependencies. 
-Dependencies changelog are stored in a Kafka topic to be be stored as materialized view on 
-Zipkin instances.
+Kafka Streams topology:
 
-### Stream processors
+![trace aggregation](docs/trace-aggregation-topology.png)
 
-#### Trace Aggregation Stream Processor
+#### Trace Store Stream
 
-This is the main processors that take incoming spans and aggregate them into:
+This component build local stores from state received on `span` Kafka topic 
+(default topic name: `zipkin-span`) for traces, service names and autocomplete tags. 
 
-- Traces
-- Dependencies
+Kafka Streams source code: [TraceStoreTopologySupplier](storage/src/main/java/zipkin2/storage/kafka/streams/TraceStoreTopologySupplier.java)
 
-![service aggregation](docs/service-aggregation-stream.png)
+Kafka Streams topology:
 
-![dependency aggregation](docs/dependency-aggregation-stream.png)
+![trace store](docs/trace-store-topology.png)
 
-#### Store Stream Processors
+#### Dependency Store
 
-Global tables for traces, service names and dependencies to be available on local state.
+This component build local store from state received on `dependency` Kafka topic (default name: `zipkin-dependency`)
 
-![trace store](docs/trace-store-stream.png)
+It builds a 1 minute time-window when counts calls and errors.
 
-![service store](docs/service-store-stream.png)
+Kafka Streams source code: [DependencyStoreTopologySupplier](storage/src/main/java/zipkin2/storage/kafka/streams/DependencyStoreTopologySupplier.java)
 
-![dependency store](docs/dependency-store-stream.png)
+Kafka Streams topology:
 
-#### Index Stream Processor
+![dependency store](docs/dependency-store-topology.png)
 
-Custom processor to full-text indexing of traces using Lucene as back-end.
+### Kafka Span Store
 
-![span index](docs/span-index-stream.png)
+This component supports search and query APIs on top of local state stores build by the Store 
+Kafka Streams component.
 
-#### Retention Stream Processor
+Component source code: [KafkaSpanStore.java](storage/src/main/java/zipkin2/storage/kafka/KafkaSpanStore.java)
 
-This is the processor that keeps track of trace timestamps for cleanup.
+#### Get Service Names/Get Span Names/Get Remote Service Names
 
-![trace retention](docs/trace-retention-stream.png)
\ No newline at end of file
+These queries are supported by service names indexed stores built from `spans` Kafka topic.
```

Comment:
somewhere above in this doc, introduce the `spans` Kafka topic, so you don't need to scroll to another doc to understand its nature. Note, elsewhere in this doc it is called `span`, so look for drift.